src/lib.rs

```diff
@@ -186,50 +193,170 @@ decl_module! {
 
 impl<T: Trait> Module<T> {
 
-    pub fn create_class(name: Vec<u8>, description: Vec<u8>, properties: Vec<Property>, schemas: Vec<ClassSchema>) -> Result<ClassId, &'static str> {
-        // TODO impl
+    pub fn create_class(
```

Comment:
I think we can make this simpler, let `properties` and `schemas` be blank by default, and force using explicit calls later. See comments on `Entity` for a similar point, its simpler, harder to use wrong, and less validation. One complex validation missing, for example, is checking that `ClassSchema` objects placed in `schemas` actually reference real existing `Properties` in `Class.properties`.

Reply:
But then, what signature do you expect from this method? Only `name` and `description`?

```rust
pub fn create_class(
        name: Vec<u8>,
        description: Vec<u8>
    ) -> Result<ClassId, &'static str> { /* ... */ }
```

and then if you want to add properties, you need to call:

```rust
pub fn add_class_property(class_id: ClassId, property: Property)
```

and then to add schemas, you need to call:

```rust
pub fn add_class_schema(class_id: ClassId, schema: ClassSchema)
```

My concerns are:
1. To fulfill a class with schemas and properties we need to send 3 transactions.
2. Is there any sense to have a class with no properties and no schemas?

Reply:
However, these functions are not extrinsics, so my concern no.1 about 2 transactions is not valid anymore.

Reply:
Refactored into two methods:
- `create_class`
- `add_class_schema`